src/runtime_src/core/edge/skd/xrt_skd.cpp

```diff
@@ -35,313 +40,370 @@ namespace xrt {
    * @param soft kernel CU index
    *
    */
-  skd::skd(xclDeviceHandle handle, int sk_meta_bohdl, int sk_bohdl, char *kname, uint32_t cu_index, unsigned char *uuid) {
-    strcpy(sk_name,kname);
-    parent_devHdl = handle;
-    cu_idx = cu_index;
-    sk_bo = sk_bohdl;
-    sk_meta_bo = sk_meta_bohdl;
-    memcpy(xclbin_uuid,uuid,sizeof(xclbin_uuid));
-
-    // Set sk_path according to sk_name
-    snprintf(sk_path, XRT_MAX_PATH_LENGTH, "%s%s%d", SOFT_KERNEL_FILE_PATH,
-	     sk_name,cu_idx);
+  skd::skd(const xclDeviceHandle handle, const int sk_meta_bohdl, const int sk_bohdl, const std::string kname, const uint32_t cu_index,
+	   unsigned char *uuid_in, const int parent_mem_bo_in, const uint64_t mem_start_paddr_in, const uint64_t mem_size_in)
+    : m_sk_name(kname), m_parent_devhdl(handle), m_cu_idx(cu_index), m_sk_bo(sk_bohdl), m_sk_meta_bo(sk_meta_bohdl),
```

Comment:
To allow for better ease of reading, please place each variable on a separate line.